src/components/container/IntradayHeartRateChart/IntradayHeartRateChart.tsx

```diff
@@ -0,0 +1,133 @@
+    timestamp: number 
+    date: Date
+    value?: number
+    rawValue?: number
+}
+
+export default function IntradayHeartRateChart(props: IntradayHeartRateChartProps) {
+    const [loading, setLoading] = useState(true);
+    const [data, setData] = useState<IntradayDataPoint[] | undefined>(undefined);
+    const [options, setOptions] = useState<MultiSeriesLineChartOptions | undefined>(undefined);
+    const dateRangeContext = useContext(DateRangeContext);
+    const intervalStart = startOfDay(dateRangeContext?.intervalStart ?? new Date());
+    const intervalEnd = startOfDay(add(intervalStart, { days: 1 }));
+    let layoutContext = useContext(LayoutContext);
+    const lineColor = resolveColor(layoutContext.colorScheme, props.lineColor) || '#bbb';
+
+    function initialize() {
+        if (props.previewState) {
+            var data: IntradayHeartRateData = {};
+            switch (props.previewState) {
+                case "Default":
+                case "CompleteDataWithThresholds":
+                    data = FullDayData;
+                    break;
+                case "PartialDataWithThresholds":
+                    data = HalfDayData;
+                    break;
+                case "MissingMidDayDataThresholds":
+                    data = MissingMidDayData;
+                    break;
+            }
+
+            transformToChartData(data);
+        } else {
+            combinedIntradayHeartRateDataProvider(
+                props.dataSources,
+                intervalStart,
+                intervalEnd,
+                props.aggregationOption,
+                props.aggregationIntervalMinutes
+            ).then((data: IntradayHeartRateData) => {
+                transformToChartData(data);
+            }).catch((error: any) => {
+                console.log("error");
+                console.log(error);
+            });
+        }
+    }
+
+    function transformToChartData(data: IntradayHeartRateData) {
+        const start = intervalStart;
+        const end = intervalEnd;
+        const interval: Interval = { start, end };
+        const options: any = { step: 5 };
+        var expectedIntervals = eachMinuteOfInterval(interval, options);
+        var chartData: any[] = expectedIntervals.map((date) => transformToDataPoint(date, data));
+
+        var yDomain: number[] = [];
+        Object.values(chartData).forEach(dataPoint => {
+            dataPoint.value !== undefined && yDomain.push(dataPoint.value);
+        });
+
+        const yMaxValue = Math.max(...yDomain);
+        const yMinValue = Math.min(...yDomain);
+
+        setOptions({ domainMin: yMinValue, domainMax: yMaxValue, thresholds: props.thresholds, connectNulls: false });
+        setData(chartData);
+        setLoading(false);
+    }
+
+    function transformToDataPoint(date: Date, data: IntradayHeartRateData) {
+        var dayKey = date.getTime();
+        var intradayDataPoint: IntradayDataPoint = { timestamp: dayKey, date: date };
```

Comment:
Minor:  My IDE is warning me that all of these `var`'s should be `let`'s or `const`'s.